lib/mysql/connector/protocol.py

```diff
@@ -475,9 +475,18 @@ def _parse_binary_values(self, fields, packet, charset='utf-8'):
             elif field[1] == FieldType.TIME:
                 (packet, value) = self._parse_binary_time(packet, field)
                 values.append(value)
+            elif field[1] in (FieldType.BLOB, FieldType.TINY_BLOB,
+                              FieldType.MEDIUM_BLOB, FieldType.LONG_BLOB):
```

Comment:
This is wrong fix.  Field type is not column type.  MEDIUM_BLOB field type is used to MEDIUM_TEXT column too.
You need to check charsetnr==63.  ([see this](https://github.com/PyMySQL/PyMySQL/blob/9dcefe9814bb053b1718a4407bb06790cb5de955/pymysql/connections.py#L1225))

When filed type is BLOB and charsetnr != 63, it is TEXT column.
When filed type is BLOB and charsetnr == 63, it is BLOB column.

Reply:
Thank you for your review.
I understand that field type BLOB include both TEXT and BLOB data type.

Currently, mysql-connector-python drop charsetnr info at parse column-packet. (https://github.com/mysql/mysql-connector-python/blob/master/lib/mysql/connector/protocol.py#L279)

I think it is too big impact to change the structure of column info tuple for this issue.
So I only remove field type checking.
Fallback processing is an alternative checking charsetnr.